aiida/engine/processes/calcjobs/tasks.py

```diff
@@ -467,6 +538,9 @@ def parse(self, retrieved_temporary_folder: str) -> plumpy.process_states.Runnin
             ProcessState.RUNNING, self.process.parse, retrieved_temporary_folder
         )  # type: ignore[return-value]
 
+    def finalize(self, exit_code):
+        return self.create_state(ProcessState.RUNNING, lambda: exit_code)
```

Comment:
My `plumpy` knowledge is probably lacking, but I don't understand why we pass the `exit_code` as a lambda function here.

Reply:
This is actually a work around to an awful hack and I will change it. Essentially `create_task` expects a callable as the first argument, which is normally the function that should be called once the `Running` state is entered again. Since I didn't really have a function to call anymore, since the process was already parsed and therefore done, I return a lambda. The lambda simply return the exit code, which is what the parsing method would have done.